controllers/configurationpolicy_controller.go

```diff
@@ -2725,14 +2718,36 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 				return false, "", false, false
 			}
 
-			// The object would have been updated, so if it's inform, return as noncompliant.
-			if strings.EqualFold(string(remediation), string(policyv1.Inform)) {
-				r.setEvaluatedObject(obj.policy, obj.existingObj, false)
+			// Generate and log the diff
+			if objectT.RecordDiff == policyv1.RecordDiffLog {
+				diff, err := generateDiff(existingObjectCopy, dryRunUpdatedObj)
+				if err != nil {
+					log.Info("Failed to generate the diff: " + err.Error())
+				} else {
+					log.Info("Logging the diff:\n" + diff)
+				}
+			}
+		} else if isInform && objectT.RecordDiff == policyv1.RecordDiffLog {
```

Comment:
As discussed in Slack, it'd be nice to have an info log message that says the diff could be shown if the new recordDiff field is set. I can't think of a nice way to specify it but maybe `spec.object-templates[*].recordDiff`  or something?

Reply:
I like that path spec. Updated!